engineioxide/src/engine.rs

```diff
@@ -210,6 +226,52 @@ where
         Ok(http_response(StatusCode::OK, "ok")?)
     }
 
+    fn parse_packets<R: BufRead + 'static>(&self, mut reader: R, protocol: ProtocolVersion) -> Result<impl Iterator<Item = Result<Packet, Error>>, String> {
```

Comment:
I think that the `parse_packet` logic should be in the packet module in the TryFromString Packet fn. It would be possible with the feature flag solution (because there would be no need to pass the `ProtocolVersion` enum)

Reply:
Ended up moving this to a new `payload.rs` module, as a payload consists of multiple packets and it made sense to me to move such a method there.